src/all/comicgrowl/src/eu/kanade/tachiyomi/extension/all/comicgrowl/ImageDescrambler.kt

```diff
@@ -0,0 +1,70 @@
+package eu.kanade.tachiyomi.extension.all.comicgrowl
+
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import android.graphics.Canvas
+import android.graphics.Rect
+import okhttp3.Interceptor
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.Response
+import okhttp3.ResponseBody.Companion.toResponseBody
+import java.io.ByteArrayOutputStream
+
+object ImageDescrambler {
+
+    // Left-top corner position
+    private class TilePos(val x: Int, val y: Int)
+
+    /**
+     * Interceptor to descramble the image.
+     */
+    fun interceptor(chain: Interceptor.Chain): Response {
+        val request = chain.request()
+        val response = chain.proceed(request)
+
+        val scramble = request.url.fragment ?: return response // return if no scramble fragment
+        val tiles = buildList {
+            scramble.split("-").forEachIndexed { index, s ->
+                val scrambleInt = s.toInt()
+                add(index, TilePos(scrambleInt / 4, scrambleInt.mod(4)))
```

Comment:
Does this not work in Kotlin?
```suggestion
                add(index, TilePos(scrambleInt / 4, scrambleInt % 4))
```